### PR TITLE
fix: Avoid memory copy in RESTful search and query

### PR DESCRIPTION
**Pull request by bigsheeper**
Custom `jsonRender` that encodes JSON data directly into the response stream, it uses less memory since it does not buffer the entire JSON structure before sending it, unlike `c.JSON` in `HTTPReturn`, which serializes the JSON fully in memory before writing it to the response.
Benchmark testing shows that, using the custom render incurs no performance loss and reduces memory consumption by nearly 50%.

issue: https://github.com/milvus-io/milvus/issues/37671

### CODE REVIEW

**Comment by bigsheeper** [comment]
Benchmark:
**10MB**
```
cpu: Intel(R) Core(TM) i7-8700 CPU @ 3.20GHz
BenchmarkHTTPReturn
BenchmarkHTTPReturn/test_HTTPReturn
BenchmarkHTTPReturn/test_HTTPReturn-12         	      87	  13127452 ns/op	27992718 B/op	      34 allocs/op
BenchmarkHTTPReturn/test_HTTPReturnStream
BenchmarkHTTPReturn/test_HTTPReturnStream-12   	      87	  12804875 ns/op	14361636 B/op	      31 allocs/op
PASS
```
**1KB**
```
cpu: Intel(R) Core(TM) i7-8700 CPU @ 3.20GHz
BenchmarkHTTPReturn
BenchmarkHTTPReturn/test_HTTPReturn
BenchmarkHTTPReturn/test_HTTPReturn-12         	  231253	      5026 ns/op	    8639 B/op	      28 allocs/op
BenchmarkHTTPReturn/test_HTTPReturnStream
BenchmarkHTTPReturn/test_HTTPReturnStream-12   	  230566	      4769 ns/op	    7582 B/op	      28 allocs/op
PASS
```

**Comment by bigsheeper** [comment]
rerun cpp-unit-test

File: internal/distributed/proxy/httpserver/handler_v2.go [hunk 1]
```diff
@@ -660,7 +676,7 @@ func (h *HandlersV2) query(ctx context.Context, c *gin.Context, anyReq any, dbNa
 				HTTPReturnMessage: merr.ErrInvalidSearchResult.Error() + ", error: " + err.Error(),
 			})
 		} else {
-			HTTPReturn(c, http.StatusOK, gin.H{
+			HTTPReturnStream(c, http.StatusOK, gin.H{
```
Comment (czs007) [inline internal/distributed/proxy/httpserver/handler_v2.go:679]: TODO: rename this if you can not disable the buffer.

**Pull request merged**